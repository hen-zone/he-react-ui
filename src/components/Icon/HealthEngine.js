// @flow
import React from 'react';

type Props = {
  inverted?: boolean,
};

export default function HealthEngine(props: Props & SVGProps) {
  const { inverted, ...restProps } = props;

  return inverted ? (
    <svg {...restProps} viewBox="0 0 48 38">
      <path
        fill="#5BC0C9"
        d="M12.895 32.745a5.689 5.689 0 0 0 11.33 0V25.16h-11.33v7.585zM24.225 6.216a5.689 5.689 0 1 0-11.33 0v7.585h11.33V6.216z"
      />
      <path
        fill="#C6E400"
        d="M37.49 25.141h-5.689v7.585a5.689 5.689 0 0 0 11.33 0V19.452a5.689 5.689 0 0 1-5.641 5.69z"
      />
      <path
        fill="#C6E400"
        d="M37.49.546a5.689 5.689 0 0 0-5.689 5.689v7.585h5.689a5.689 5.689 0 0 1 5.689 5.69V6.234A5.689 5.689 0 0 0 37.49.546z"
      />
      <path
        fill="#FFF"
        d="M5.17 14.043a5.689 5.689 0 0 0 0 11.33h7.585v-11.33H5.17zM24.225 13.82h7.605v11.33h-7.605z"
      />
      <path fill="#008EC8" d="M12.895 13.82h11.33v11.33h-11.33z" />
      <path
        fill="#48B400"
        d="M37.764 13.82h-5.963v11.321h5.963c3.047-.292 5.367-2.739 5.367-5.66 0-2.922-2.32-5.369-5.367-5.66z"
      />
    </svg>
  ) : (
    <svg {...restProps} viewBox="0 0 48 38">
      <path
        d="M12.908 32.745a5.692 5.692 0 0 0 5.67 5.17 5.692 5.692 0 0 0 5.672-5.17V25.16H12.908v7.585zM24.25 6.216A5.687 5.687 0 0 0 21.575.859a5.7 5.7 0 0 0-5.992 0 5.687 5.687 0 0 0-2.675 5.357v7.585H24.25V6.216z"
        fill="#5BC0C9"
      />
      <path
        d="M37.118 25.141h-5.695v7.585a5.692 5.692 0 0 0 5.671 5.17 5.692 5.692 0 0 0 5.671-5.17V19.452a5.692 5.692 0 0 1-5.647 5.69z"
        fill="#C6E400"
      />
      <path
        d="M37.118.546a5.692 5.692 0 0 0-5.695 5.689v7.585h5.695a5.692 5.692 0 0 1 5.694 5.69V6.234A5.692 5.692 0 0 0 37.118.546z"
        fill="#C6E400"
      />
      <path
        d="M5.296 13.82a5.691 5.691 0 0 0-5.175 5.665 5.691 5.691 0 0 0 5.175 5.666h7.593V13.82H5.296zM24.25 13.82h7.2v11.33h-7.2z"
        fill="#2B4B5A"
      />
      <path fill="#008EC8" d="M12.908 13.82H24.25v11.33H12.908z" />
      <path
        d="M37.528 13.82h-6.095v11.321h6.495c2.91-.292 4.726-2.739 4.926-5.66 0-2.922-2.016-5.369-4.926-5.66h-.4z"
        fill="#48B400"
      />
    </svg>
  );
}
