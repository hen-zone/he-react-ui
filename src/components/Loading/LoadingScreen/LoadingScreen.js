import React from 'react'
import PropTypes from 'prop-types'
import style from './LoadingScreen.module.scss'

export default class LoadingScreen extends React.Component {
  static propTypes = {
    children: PropTypes.element
  }

  static defaultProps = {
    children: null
  }

  render () {
    const { children } = this.props
    return (
      <div className={style.loadingContainer}>
        <div className={style.pulseButton} />
        <div className={style.pulseButton2}>
          {children}
        </div>
      </div>
    )
  }
}
