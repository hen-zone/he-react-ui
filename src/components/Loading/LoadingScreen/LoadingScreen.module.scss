.loadingContainer {
  background: rgba(0, 0, 0, 0.6);
  position: fixed;
  overflow-y: scroll;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  z-index: 1000;
}

.pulseButton,
.pulseButton2 {
  position: absolute;
  top: 50%;
  left: 50%;
  margin-left: -50px;
  margin-top: -50px;
  display: flex;
  justify-content: center;
  align-items: center;
  width: 120px;
  height: 120px;
  font-size: 1.3em;
  font-weight: light;
  font-family: "Trebuchet MS", sans-serif;
  text-transform: uppercase;
  text-align: center;
  line-height: 100px;
  letter-spacing: -1px;
  color: white;
  border: none;
  border-radius: 50%;
  background: transparent;
  cursor: pointer;
  box-shadow: 0 0 0 0 rgba(#efefef, 0.5);
  padding-right: 10px;
  box-sizing: border-box;
  background: #fafafa;
  opacity: 0.8;
}
.pulseButton:hover {
  -webkit-animation: none;
}
.pulseButton {
  -webkit-animation: pulse 1.5s infinite;
}
.pulseButton2 {
  -webkit-animation: pulse2 1.5s infinite;
}

@-webkit-keyframes pulse {
  0% {
  }
  40% {
    box-shadow: 0 0 0 0 rgba(#eeeeee, 0.5);
  }
  75% {
    box-shadow: 0 0 0 70px rgba(#eeeeee, 0);
  }
  90% {
    box-shadow: 0 0 0 0 rgba(#eeeeee, 0);
  }
}
@-webkit-keyframes pulse2 {
  0% {
    transform: scale(0.9);
  }
  60% {
    transform: scale(1);
    box-shadow: 0 0 0 50px rgba(#eeeeee, 0);
  }
  80% {
    transform: scale(0.9);
    box-shadow: 0 0 0 0 rgba(#eeeeee, 0);
  }
}
