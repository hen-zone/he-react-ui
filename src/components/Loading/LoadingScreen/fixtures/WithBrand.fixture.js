import React from 'react'

import LoadingScreen from '../LoadingScreen'
import HealthEngine from '../../../Icon/HealthEngine'
import style from './style.scss'

export default {
  component: LoadingScreen,
  props: { children: <HealthEngine className={style.loadingIcon} /> }
}
