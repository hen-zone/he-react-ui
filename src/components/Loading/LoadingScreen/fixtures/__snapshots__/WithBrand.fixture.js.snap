// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`anonymous 1`] = `
<LoadingScreen>
  <div
    className="loadingContainer"
  >
    <div
      className="pulseButton"
    />
    <div
      className="pulseButton2"
    >
      <HealthEngine
        className="loadingIcon"
      >
        <svg
          className="loadingIcon"
          viewBox="0 0 48 38"
        >
          <path
            d="M12.908 32.745a5.692 5.692 0 0 0 5.67 5.17 5.692 5.692 0 0 0 5.672-5.17V25.16H12.908v7.585zM24.25 6.216A5.687 5.687 0 0 0 21.575.859a5.7 5.7 0 0 0-5.992 0 5.687 5.687 0 0 0-2.675 5.357v7.585H24.25V6.216z"
            fill="#5BC0C9"
          />
          <path
            d="M37.118 25.141h-5.695v7.585a5.692 5.692 0 0 0 5.671 5.17 5.692 5.692 0 0 0 5.671-5.17V19.452a5.692 5.692 0 0 1-5.647 5.69z"
            fill="#C6E400"
          />
          <path
            d="M37.118.546a5.692 5.692 0 0 0-5.695 5.689v7.585h5.695a5.692 5.692 0 0 1 5.694 5.69V6.234A5.692 5.692 0 0 0 37.118.546z"
            fill="#C6E400"
          />
          <path
            d="M5.296 13.82a5.691 5.691 0 0 0-5.175 5.665 5.691 5.691 0 0 0 5.175 5.666h7.593V13.82H5.296zM24.25 13.82h7.2v11.33h-7.2z"
            fill="#2B4B5A"
          />
          <path
            d="M12.908 13.82H24.25v11.33H12.908z"
            fill="#008EC8"
          />
          <path
            d="M37.528 13.82h-6.095v11.321h6.495c2.91-.292 4.726-2.739 4.926-5.66 0-2.922-2.016-5.369-4.926-5.66h-.4z"
            fill="#48B400"
          />
        </svg>
      </HealthEngine>
    </div>
  </div>
</LoadingScreen>
`;
